From 7e174c531b0fe98b44640f15e05828ca01886c93 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 15 Jan 2016 03:47:05 -0500
Subject: [PATCH] wr-policy-fix-security_validate_transition

Make init_t MLS trusted for lowering/raising the level of
files.

Fix security_validate_transition issues:

  op=security_validate_transition seresult=denied \
  oldcontext=system_u:object_r:device_t:s15:c0.c1023 \
  newcontext=system_u:object_r:device_t:s0 \
  taskcontext=system_u:system_r:init_t:s0-s15:c0.c1023 \
  tclass=dir

  op=security_validate_transition seresult=denied \
  oldcontext=system_u:object_r:var_run_t:s0 \
  newcontext=system_u:object_r:var_log_t:s0-s15:c0.c1023 \
  taskcontext=system_u:system_r:init_t:s0-s15:c0.c1023 \
  tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/init.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/system/init.te b/policy/modules/system/init.te
index ca8678b..5c8aff0 100644
--- a/policy/modules/system/init.te
+++ b/policy/modules/system/init.te
@@ -1431,3 +1431,7 @@ allow initrc_t init_t:system { start status reboot };
 allow initrc_t init_var_run_t:service { start status };
 
 allow initrc_t init_var_run_t:service stop;
+
+# MLS trusted for lowering/raising the level of files
+mls_file_downgrade(init_t)
+mls_file_upgrade(init_t)
-- 
2.7.4

